classes/controller/FrontController.php

```diff
@@ -459,10 +459,9 @@ public function init()
             CartRule::autoAddToCart($this->context);
         } else {
             $this->context->cart = $cart;
+            $this->context->cart->checkAndUpdateAddresses();
```

Comment:
No sure about that because it's may prevent us some pb we have on cart in joliApp where product are not removable